src/main/java/org/apache/sling/testing/mock/jcr/MockNode.java

```diff
@@ -368,8 +372,26 @@ public boolean equals(Object obj) {
 
     @Override
     public NodeType[] getMixinNodeTypes() throws RepositoryException {
-        // we have no real mixin support - just assume no mixin nodetypes are set
-        return new NodeType[0];
+        Value[] mixinNames = getProperty(JcrConstants.JCR_MIXINTYPES).getValues();
+
+        return Arrays.stream(mixinNames)
+                .map(value -> {
+                    try {
+                        return value.getString();
+                    } catch (RepositoryException e) {
+                        return new NodeType[0];
```

Comment:
i assume this should return null in case of error - to be filtered out in the next step?

Reply:
I more inclined to return an empty array because it is a mock implementation and it is safer to not return null and cause errors, but it is your call :)